docs/guides/multisite_custom.rst

```diff
@@ -0,0 +1,230 @@
+    from feincms3_meta.utils import meta_tags
+    from feincms3_sites.models import current_site
+
+    from .models import Page
+    from .renderer import renderer
+
+
+    def page_detail(request, path=None):
+        """
+        The view expects uri paths to be formed as such
+
+        [[/<language-code>][/<path>]]
+
+        in case of an empty uri path the view will redirect the user to
+
+        /<site.default_language/
+
+        expecting a root page to exist that has the appropriate path
+
+        otherwise, the provided uri path will be used and the system will
+        try to retrieve a probably existing page via the ORM.
+        """
+        actual_site = current_site()
+        if not path:
+            # make sure that the user is redirected to the correct url
+            # we expect a root page with custom path /<default_language>/ to exist here
+            redirect_path = "/{}/".format(actual_site.default_language)
+            return redirect(redirect_path)
+
+        actual_path = "/{}/".format(path)
+        page = get_object_or_404(
+            Page.objects.active(),
+            path=actual_path
+        )
+        page.activate_language(request)
+        ancestors = page.ancestors().reverse()
+        meta_data_providers = [page] + list(ancestors) + [page.site]
+        return render(
+            request,
+            page.type.template_name,
+            {
+                "page": page,
+                "regions": Regions.from_item(
+                    page,
+                    renderer=renderer,
+                    inherit_from=ancestors,
+                    timeout=60
+                ),
+                "meta_tags": meta_tags(
+                    meta_data_providers,
+                    request=request,
+                    site_name=page.site.name
+                )
+            },
+        )
+
+Make sure that your custom site model get registered with the Django ORM
+before your page model gets registered, otherwise there will be an exception
+telling you that the site model configured in ``FEINCMS3_SITES_SITE_MODEL``
+does not exist.
+
+Remember to update your migrations as well:
+
+.. code-block:: shell
+
+    ./manage.py makemigrations
+    ./manage.py migrate
+
+
+Implementing a Custom Site Model Admin
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from django import forms
+    from django.conf import settings
+    from django.contrib import admin
+    from django.utils.translation import gettext_lazy as _
+
+    from feincms3_meta.models import MetaMixin
+
+    from .models import Site
+
+
+    class SiteForm(forms.ModelForm):
+        default_language = Site._meta.get_field("default_language").formfield(
+            choices=[("", "----------")] + list(settings.LANGUAGES),
+        )
+
+        class Meta:
+            model = Site
+            fields = "__all__"
+
+
+    class SiteDefaultLanguageListFilter(admin.SimpleListFilter):
```

Comment:
Is this necessary because you don't want to display the full list of languages `global_settings.LANGUAGES`? If yes, a short comment would be great!

Reply:
Hm, I have copied this over from feincms3-sites. feincms3-sites will declare site model with a default_language field that depends on global_settings.LANGUAGES in order to prevent it from generating spurious django orm migrations on changes to that field.

in your feincms3-sites you will then limit the available options to the actual settings.LANGUAGE setting.
Here, we will only add a list filter that will support that limitation, otherwise we would have all of global_settings.LANGUAGE, which we do not want.

Point taken, will add a short comment.

Reply:
I have added  the comment

``` 
# filter languages by settings.LANGUAGE and not by global_settings.LANGUAGE
class SiteDefaultLanguageListFilter(admin.SimpleListFilter):
...
```